File: ac-predictor-extension/src/libs/contest/results/result.ts
export class Result {
    IsRated: boolean;
    IsSubmitted: boolean;
    UserScreenName: string;
    Place: number;
    RatedRank: number;
    OldRating: number;
    NewRating: number;
    Competitions: number;
    Performance: number;
    InnerPerformance: number;
    constructor(
        isRated: boolean,
        isSubmitted: boolean,
        userScreenName: string,
        place: number,
        ratedRank: number,
        oldRating: number,
        newRating: number,
        competitions: number,
        performance: number,
        innerPerformance: number
    ) {
        this.IsRated = isRated;
        this.IsSubmitted = isSubmitted;
        this.UserScreenName = userScreenName;
        this.Place = place;
        this.RatedRank = ratedRank;
        this.OldRating = oldRating;
        this.NewRating = newRating;
        this.Competitions = competitions;
        this.Performance = performance;
        this.InnerPerformance = innerPerformance;
    }
}

File: crawler/ac_predictor_crawler/domain/result.py
from dataclasses import dataclass
from typing import Dict, Optional

def parse_int(s: str, default: int):
  if len(s) != 0:
    if not s.isdigit(): raise Exception("invalid format")
    return int(s)
  return default

@dataclass
class Result:
  is_rated: bool
  place: int
  old_rating: int
  new_rating: int
  performance: int
  user_screen_name: str

  inner_performance: Optional[int]

  def to_dict(self):
    res = {
      "IsRated": self.is_rated,
      "Place": self.place,
      "OldRating": self.old_rating,
      "NewRating": self.new_rating,
      "Performance": self.performance,
      "UserScreenName": self.user_screen_name,
    }
    if self.inner_performance is not None:
      res["InnerPerformance"] = self.inner_performance
    return res

  @staticmethod
  def from_dict(s: Dict):
    return Result(
      s["IsRated"],
      s["Place"],
      s["OldRating"],
      s["NewRating"],
      s["Performance"],
      s["UserScreenName"],
      s.get("InnerPerformance")
    )
